### PR TITLE
enable in dropwizard 1.0.0

### PR DESCRIPTION
**Pull request by SerSchi**
Hi @stve 

i´ve made some changes to enable raven in the latest dropwizard release (1.0.0), maybe you are interested in this.

regards
Sergej

### CODE REVIEW

**Comment by mickeyreiss** [comment]
👍, but it would be useful for me if this were released in a separate version of dropwizard-raven from #10.

**Comment by stve** [comment]
I may switch the approach around hard-coded dependencies, especially now that DW is 1.0. In theory, we could use maven `provided` keyword vs. fixing releases to specific Dropwizard releases. Presumably the same could be accomplished for most Raven releases as well.

**Comment by stve** [comment]
@SerSchi thanks for this pull request. The formatting of some of the code in here was a mess and I've just gone through and cleaned it all up. Would you mind rebasing your changes onto the latest commits in master? That should cut down size of the diff as well. Thanks!

**Comment by SerSchi** [comment]
Ok, i'll do it tomorrow. 
Am 13.08.2016 06:16 schrieb Steve Agalloco notifications@github.com:@SerSchi thanks for this pull request. The formatting of some of the code in here was a mess and I've just gone through and cleaned it all up. Would you mind rebasing your changes onto the latest commits in master? That should cut down size of the diff as well. Thanks!

**Comment by SerSchi** [comment]
done!

**Comment by stve** [comment]
hi @SerSchi 

Thank you for updating your pull request. I am seeing quite a bit of formatting changes in the diff which are 1) reformatting nearly all of the source code and 2) making it difficult for me to figure out what is changing.

Would you mind taking a stab at addressing the formatting so that I can give this a proper review? Thanks!

File: pom.xml [hunk 1]
```diff
@@ -59,8 +59,8 @@
     </issueManagement>
 
     <properties>
-        <io.dropwizard.version>0.9.2</io.dropwizard.version>
         <getsentry.raven.version>7.6.0</getsentry.raven.version>
+        <io.dropwizard.version>1.0.0</io.dropwizard.version>
```
Comment (SerSchi) [inline pom.xml:63]: need to add for DW-1.0.0

File: src/main/java/com/tradier/raven/logging/RavenAppenderFactory.java [hunk 2]
```diff
@@ -5,9 +5,11 @@
 import ch.qos.logback.classic.LoggerContext;
 import ch.qos.logback.classic.spi.ILoggingEvent;
 import ch.qos.logback.core.Appender;
-import ch.qos.logback.core.Layout;
 import ch.qos.logback.core.filter.Filter;
 import ch.qos.logback.core.spi.FilterReply;
+import io.dropwizard.logging.async.AsyncAppenderFactory;
+import io.dropwizard.logging.filter.LevelFilterFactory;
+import io.dropwizard.logging.layout.LayoutFactory;
```
Comment (SerSchi) [inline src/main/java/com/tradier/raven/logging/RavenAppenderFactory.java:12]: need to add for DW-1.0.0

File: src/main/java/com/tradier/raven/logging/RavenAppenderFactory.java [hunk 3]
```diff
@@ -16,83 +18,84 @@
...
+            appender.setTags(tags);
+        }
+
+        appender.start();
+        appender.addFilter(levelFilterFactory.build(threshold));
+        final Appender<ILoggingEvent> asyncAppender = wrapAsync(appender, asyncAppenderFactory, context);
+        addDroppingRavenLoggingFilter(asyncAppender);
+
+        return asyncAppender;
+    }
```
Comment (SerSchi) [inline src/main/java/com/tradier/raven/logging/RavenAppenderFactory.java:82]: need to change for DW-1.0.0 => build

File: src/main/java/com/tradier/raven/logging/RavenBootstrap.java [hunk 4]
```diff
@@ -6,72 +6,89 @@
 import ch.qos.logback.classic.Logger;
 
 import com.google.common.base.Optional;
+import io.dropwizard.logging.async.AsyncLoggingEventAppenderFactory;
+import io.dropwizard.logging.filter.ThresholdLevelFilterFactory;
+import io.dropwizard.logging.layout.DropwizardLayoutFactory;
```
Comment (SerSchi) [inline src/main/java/com/tradier/raven/logging/RavenBootstrap.java:12]: need to add for DW-1.0.0

Comment (SerSchi) [inline src/main/java/com/tradier/raven/logging/RavenBootstrap.java:12]: need to add for DW-1.0.0

File: src/main/java/com/tradier/raven/logging/RavenBootstrap.java [hunk 5]
```diff
@@ -6,72 +6,89 @@
...
+     *
+     * @param dsn The DSN (Data Source Name) for your project
+     * @param tags Custom tags to send to Sentry along with errors
+     * @param cleanRootLogger If true, detach and stop all other appenders from the root logger
+     */
+    public static void bootstrap(final String dsn, final Optional<String> tags, final boolean cleanRootLogger) {
+        final RavenAppenderFactory raven = new RavenAppenderFactory();
+        raven.setThreshold(Level.ERROR);
+        raven.setDsn(dsn);
+        raven.setTags(tags.get());
```
Comment (SerSchi) [inline src/main/java/com/tradier/raven/logging/RavenBootstrap.java]: need to change for DW-1.0.0 => bootstrap

File: src/main/java/com/tradier/raven/logging/RavenBootstrap.java [hunk 6]
```diff
@@ -6,72 +6,89 @@
...
-      root.detachAndStopAllAppenders();
+        if (cleanRootLogger) {
+            root.detachAndStopAllAppenders();
+        }
+        final ThresholdLevelFilterFactory levelFilterFactory = new ThresholdLevelFilterFactory();
+        final DropwizardLayoutFactory layoutFactory = new DropwizardLayoutFactory();
+        final AsyncLoggingEventAppenderFactory asyncAppenderFactory = new AsyncLoggingEventAppenderFactory();
+        root.addAppender(
+                raven.build(root.getLoggerContext(), dsn, layoutFactory, levelFilterFactory, asyncAppenderFactory));
     }
```
Comment (SerSchi) [inline src/main/java/com/tradier/raven/logging/RavenBootstrap.java:93]: need to change for DW-1.0.0 => another initializing root-Appender

File: src/test/java/com/tradier/raven/logging/RavenAppenderFactoryTest.java [hunk 7]
```diff
@@ -13,6 +13,9 @@
 import ch.qos.logback.classic.LoggerContext;
 import ch.qos.logback.classic.spi.ILoggingEvent;
 import ch.qos.logback.core.Appender;
+import io.dropwizard.logging.async.AsyncLoggingEventAppenderFactory;
+import io.dropwizard.logging.filter.ThresholdLevelFilterFactory;
+import io.dropwizard.logging.layout.DropwizardLayoutFactory;
```
Comment (SerSchi) [inline src/test/java/com/tradier/raven/logging/RavenAppenderFactoryTest.java:19]: need to add for DW-1.0.0

File: src/test/java/com/tradier/raven/logging/RavenAppenderFactoryTest.java [hunk 8]
```diff
@@ -26,15 +29,19 @@ public void hasValidDefaults() throws IOException, ConfigurationException {
 	
     @Test(expected = NullPointerException.class)
     public void buildRavenAppenderShouldFailWithNullContext() {
-        new RavenAppenderFactory().build(null, "", null);
+        new RavenAppenderFactory().build(null, "", null, null, null);
     }
```
Comment (SerSchi) [inline src/test/java/com/tradier/raven/logging/RavenAppenderFactoryTest.java:33]: need to change for DW-1.0.0

File: src/test/java/com/tradier/raven/logging/RavenAppenderFactoryTest.java [hunk 9]
```diff
@@ -26,15 +29,19 @@ public void hasValidDefaults() throws IOException, ConfigurationException {
...
     public void buildRavenAppenderShouldWorkWithValidConfiguration() {
         final RavenAppenderFactory raven = new RavenAppenderFactory();
         final String dsn = "https://user:pass@app.getsentry.com/id";
 
-        Appender<ILoggingEvent> appender = raven.build(new LoggerContext(), dsn, null);
+        ThresholdLevelFilterFactory levelFilterFactory = new ThresholdLevelFilterFactory();
+        DropwizardLayoutFactory layoutFactory = new DropwizardLayoutFactory();
+        AsyncLoggingEventAppenderFactory asyncAppenderFactory = new AsyncLoggingEventAppenderFactory();
+        Appender<ILoggingEvent> appender = raven.build(
+                new LoggerContext(), dsn, layoutFactory, levelFilterFactory, asyncAppenderFactory);
```
Comment (SerSchi) [inline src/test/java/com/tradier/raven/logging/RavenAppenderFactoryTest.java:45]: need to change for DW-1.0.0

**Comment by SerSchi** [comment]
Hi Steve,

i made some inline-comments, i hope it will help you to find out the 
real changes. if not please fill free and write me.

**Comment by stve** [comment]
Hi @SerSchi, I had difficultly merging your changes so I pulled them into a new branch. Please review: https://github.com/tradier/dropwizard-raven/compare/1.0.0

**Comment by SerSchi** [comment]
LGTM

**Comment by jennifergarcia** [comment]
Is this going to be released anytime soon?  This is the last thing we are waiting on for upgrading Dropwizard.

Thanks,
Jenny

**Comment by stve** [comment]
I've just released version `1.0.0-1` and it should be available in the central repository shortly. Sorry for the delay.

**Pull request closed without merging**